### ISSUE TITLE
Add a way to expose statistics to Metrics

### ISSUE THREAD

**Issue by michalszynkiewicz** [smallrye/smallrye-stork#63]
Micrometer integration, probably through an abstraction layer, so that other metrics engines can be integrate too.

- service instances resolved
- service discovery executions
- load balancing selection metrics

**Comment by michalszynkiewicz** [comment]
Maybe: event based metrics based on JFR, than collected and pushed to Micrometer

**Comment by cescoffier** [comment]
Followed up by https://github.com/quarkusio/quarkus-upstream-roadmap/issues/21.

```yaml
organization: QuarkusIO
repository: quarkus-upstream-roadmap
issue: 21
url: https://github.com/quarkusio/quarkus-upstream-roadmap/issues/21
```

**Comment by cescoffier** [comment]
We may want to use micrometer, and the micrometer observation API directly.

**Comment by aureamunoz** [comment]
Fixed by #681
